java-components/build-recipes-database/src/main/java/com/redhat/hacbs/recipies/scm/GitScmLocator.java

```diff
@@ -141,7 +141,14 @@ private RecipeGroupManager getRecipeGroupManager() {
                 if (remotePattern == null || remotePattern.matcher(i).matches()) {
                     log.infof("Cloning recipe repo %s", i);
                     try {
-                        repoManager = RecipeRepositoryManager.create(i, "main", Optional.empty(),
+                        // Allow cloning of another branch via <url>#<branch> format.
+                        String branch = "main";
```

Comment:
This is used in other places as well. Lookup all uses of the `RecipeRepositoryManager.create` function.

It would probably be better if this was pushed down into `RecipeRepositoryManager`